flash/image/classification/adapters.py

```diff
@@ -0,0 +1,416 @@
+        batch_size: int,
+        num_workers: int,
+        pin_memory: bool,
+        collate_fn: Callable,
+        shuffle: bool,
+        drop_last: bool,
+        sampler: Optional[Sampler],
+    ) -> DataLoader:
+        assert batch_size == 1
+        return super().process_train_dataset(
+            self.convert_dataset(dataset),
+            batch_size,
+            num_workers,
+            pin_memory,
+            collate_fn,
+            shuffle=shuffle,
+            drop_last=drop_last,
+            sampler=sampler,
+        )
+
+    def process_val_dataset(
+        self,
+        dataset: BaseAutoDataset,
+        batch_size: int,
+        num_workers: int,
+        pin_memory: bool,
+        collate_fn: Callable,
+        shuffle: bool = False,
+        drop_last: bool = False,
+        sampler: Optional[Sampler] = None,
+    ) -> DataLoader:
+        assert batch_size == 1
+        return super().process_val_dataset(
+            self.convert_dataset(dataset),
+            batch_size,
+            num_workers,
+            pin_memory,
+            collate_fn,
+            shuffle,
+            drop_last,
+            sampler,
+        )
+
+    def process_test_dataset(
+        self,
+        dataset: BaseAutoDataset,
+        batch_size: int,
+        num_workers: int,
+        pin_memory: bool,
+        collate_fn: Callable,
+        shuffle: bool = False,
+        drop_last: bool = False,
+        sampler: Optional[Sampler] = None,
+    ) -> DataLoader:
+        assert batch_size == 1
+        return super().process_test_dataset(
+            self.convert_dataset(dataset),
+            batch_size,
+            num_workers,
+            pin_memory,
+            collate_fn,
+            shuffle,
+            drop_last,
+            sampler,
+        )
+
+    def process_predict_dataset(
+        self,
+        dataset: BaseAutoDataset,
+        batch_size: int = 1,
+        num_workers: int = 0,
+        pin_memory: bool = False,
+        collate_fn: Callable = lambda x: x,
+        shuffle: bool = False,
+        drop_last: bool = True,
+        sampler: Optional[Sampler] = None,
+    ) -> DataLoader:
+        assert batch_size == 1
+
+        if not self._algorithm_has_validated:
+            raise MisconfigurationException(
+                "This training_strategies requires to be validated. Call trainer.validate(...)."
+            )
+
+        return super().process_predict_dataset(
+            dataset,
+            batch_size,
+            num_workers,
+            pin_memory,
+            collate_fn,
+            shuffle,
+            drop_last,
+            sampler,
+        )
+
+
+class DefaultAdapter(Adapter):
+    """The ``DefaultAdapter`` is an :class:`~flash.core.adapter.Adapter`."""
+
+    required_extras: str = "image"
+
+    def __init__(self, task: AdapterTask, backbone: torch.nn.Module, head: torch.nn.Module):
+        super().__init__()
+
+        self._task = NoModule(task)
```

Comment:
Could we just move out the base `*_step` implementations to a mixin to just inherit the behaviour rather than have a reference to the task?

Reply:
Not sure to understand you. I did it this way, so we can use `Task.training_step(self.task, batch, batch_idx)` in the training_step of the adapter.

Reply:
Yeah, but we could just move the Task.*_step methods out somehwere so that they can be included in the base task but also in adapters. That way the base adapter and base task could both have the base step methods and you could just do `super().*_step` to get access to it.